crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -59,15 +59,38 @@ def services
     if request.post?
       service_object = CrowbarService.new(Rails.logger)
 
-      # prepare the scripts for various actions in one chef run
+      # First, prepare the scripts for various actions in one chef run
       service_object.prepare_nodes_for_os_upgrade
 
-      # now, initiate the services shutdown by calling scripts on all nodes
-      # FIXME: this needs to go elsewhere
-      # FIXME: this needs to be orchestrated one by one
+      # FIXME: This is primitive orchestration code that should be written elsewhere
+      # (library/model/external tool)
+
+      # Second, initiate the services shutdown in all clusters
+      # These are the non essential services that could be shut down everywhere
+      # FIXME: non-HA case not being handled
       NodeObject.find("pacemaker_founder:true").each do |node|
```

Comment:
probably we don't need this anymore since the loop you introduced below goes through all cluster members, including the founder one. I don't think we do anything special for the founder one.

Maybe later on we want to choose a specific node to start the shutdown and upgrade sequence, i.e. starting with the DRBD slave. But that can be arranged later.